shared/constants/types/fs.js

```diff
@@ -53,6 +53,17 @@ export type _PathUserSetting = {
 }
 export type PathUserSetting = I.RecordOf<_PathUserSetting>
 
+export type _TransferState = {
+  isUpload: boolean, // true if upload; false if download
+  isDir: boolean,
+  path: Path,
+  localPath: string,
```

Comment:
you probably want localPath to be an opaque type also so we don't confuse these strings